### DISCUSSION TITLE
How to get intelisense while block developing?

### DISCUSSION THREAD

**Discussion by cawa-93** [WordPress/gutenberg#49194] [Developer Experience]
I just started learning blocks and making my first block. (Off topic, but I've never worked with react before).

I immediately ran into a problem: there is no intelisense in the project.

![зображення](https://user-images.githubusercontent.com/1662812/226301076-3e4b3d22-b2af-4fd3-8a6f-6c1440e2c2c7.png)
![зображення](https://user-images.githubusercontent.com/1662812/226301212-10e5a05f-f20b-4416-969a-522b0d55f4e1.png)

My understanding is that these packages are not actually installed in dev env and are expected in the runtime only.

**Comment by johnhooks** [comment]
Is `@wordpress/block-editor` listed in your `package.json` and have you installed it?

Edit:

Also `@wordpress/block-editor` does not have its own types. You may also need to also install `@types/wordpress__block-editor` to have the experience you expect.

**Reply by cawa-93** [reply]
Yeap. This seems help. Does `@wordpress/block-editor` installed loccaly somehow affect building?

**Reply by johnhooks** [reply]
Not to my knowledge, `wp-scripts` does some magic and links `import { ... } from '@wordpress/blocks'` to the global `window.wp`, but having it installed locally is how Intellisense, and all the build tooling used in JS development, analyzes the code.
